M2/Macaulay2/m2/Makefile.tests

```diff
@@ -1,24 +0,0 @@
-# -*- Mode: Makefile -*-
```

Comment:
I don't believe this could have worked for quite a few years at least. However, with the changes in this PR (in particular the `needs` lines) you can run M2 with `--no-core` and then load individual files in the Core and I've been working on making this reliable. There's some more work to be done here later, but this test doesn't make sense.